core/services/ocrbootstrap/delegate.go

```diff
@@ -56,39 +54,24 @@ func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.ServiceCtx, e
 	if spec == nil {
 		return nil, errors.Errorf("Bootstrap.Delegate expects an *job.BootstrapSpec to be present, got %v", jobSpec)
 	}
-
-	ocr2Provider, err := d.relayer.NewOCR2Provider(jobSpec.ExternalJobID, &relay.OCR2ProviderArgs{
-		ID:              spec.ID,
-		ContractID:      spec.ContractID,
-		TransmitterID:   null.String{},
-		Relay:           spec.Relay,
-		RelayConfig:     spec.RelayConfig,
-		IsBootstrapPeer: true,
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "error calling 'relayer.NewOCR2Provider'")
-	}
-	services = append(services, ocr2Provider)
-
-	configDB := NewDB(d.db.DB, spec.ID, d.lggr)
-	peerWrapper := d.peerWrapper
-	if peerWrapper == nil {
+	if d.peerWrapper == nil {
 		return nil, errors.New("cannot setup OCR2 job service, libp2p peer was missing")
-	} else if !peerWrapper.IsStarted() {
+	} else if !d.peerWrapper.IsStarted() {
 		return nil, errors.New("peerWrapper is not started. OCR2 jobs require a started and running peer. Did you forget to specify P2P_LISTEN_PORT?")
 	}
-
-	loggerWith := d.lggr.Named("OCR").With(
-		"contractID", spec.ContractID,
-		"jobName", jobSpec.Name.ValueOrZero(),
-		"jobID", jobSpec.ID,
-	)
-	ocrLogger := logger.NewOCRWrapper(loggerWith, true, func(msg string) {
-		d.lggr.ErrorIf(d.jobORM.RecordError(jobSpec.ID, msg), "unable to record error")
+	relayer, exists := d.relayers[spec.Relay]
+	if !exists {
+		return nil, errors.Errorf("%s relay does not exist is it enabled?", spec.Relay)
+	}
+	configWatcher, err := relayer.NewConfigWatcher(types.ConfigWatcherArgs{
+		JobID:       spec.ID,
+		ContractID:  spec.ContractID,
+		RelayConfig: spec.RelayConfig.Bytes(),
```

Comment:
Hmm, before `RelayConfig` was part of the `OCR2Provider` which is a broader abstraction vs. the new `ConfigWatcherArgs`. 

For example, on Solana we had the `transmissionsID` as part of the unstructured `RelayConfig` and that parameter wasn't used directly for OCR2 config watching.

It will still work, just observing how the new abstraction reads and fits.

Reply:
yeah one drawback of parsing the relay config in the relays is it's possible that there are fields passed into the config watcher which are unused and only used in the median plugin. But IMO that's definitely worth it to help keep chain specific libraries/code out of core